cmd/state-tray/internal/open/open_lin.go

```diff
@@ -28,3 +28,38 @@ func Prompt(command string) error {
 
 	return nil
 }
+
+func preferredShellWithFallback(fallback string) (string, error) {
+	currentUser, err := user.Current()
+	if err != nil {
+		return "", locale.WrapError(err, "err_user_unknown", "Cannot get current user")
+	}
+
+	f, err := os.Open("/etc/passwd")
+	if err != nil {
+		return "", locale.WrapError(err, "err_open_passwd", "Cannot open passwd file")
```

Comment:
This should explain why we're looking at the passwd file, otherwise this is a very scary error to get.

Reply:
I changed the error message to not be so alarming.

Reply:
But shouldn't we return the fallback anyways in that case?  I don't think the user should ever be bothered with that...